### PR TITLE
Fixes cablecuff making producing 0 amount cables

### PR DESCRIPTION
**Pull request by Arturlang**
Coding is hard, okay?

No changelog because very quick bugfix i dont want a stain on my record please oh god

### CODE REVIEW

**Review by Putnam3145 — APPROVED** [review]

File: code/modules/power/cable.dm [hunk 1]
```diff
@@ -559,14 +559,19 @@ By design, d1 is the smallest direction and d2 is the highest
...
-		to_chat(user, "<span class='notice'>You make some restraints out of cable</span>")
+	if(!do_after(user, 30, TRUE, user, TRUE))
+		to_chat(user, "<span class='notice'>You fail to make cable restraints, you need to stand still while doing so.</span>")
 		return
-	to_chat(user, "<span class='notice'>You fail to make cable restraints, you need to stand still while doing so.</span>")
+	if(!check_cable_amount())
+		return
+	amount -= 15
+	if(amount == 0)
+		qdel(src)
```
Comment (Ghommie) [inline code/modules/power/cable.dm]: Why not use the `use` stack proc instead?

Comment (Arturlang) [reply code/modules/power/cable.dm]: Ill see if it works

File: code/modules/power/cable.dm [hunk 2]
```diff
@@ -559,14 +559,17 @@ By design, d1 is the smallest direction and d2 is the highest
...
-		user.put_in_hands(result)
-		result.color = color 
-		to_chat(user, "<span class='notice'>You make some restraints out of cable</span>")
+	if(!do_after(user, 30, TRUE, user, TRUE))
+		to_chat(user, "<span class='notice'>You fail to make cable restraints, you need to stand still while doing so.</span>")
 		return
-	to_chat(user, "<span class='notice'>You fail to make cable restraints, you need to stand still while doing so.</span>")
+	if(!check_cable_amount())
+		return
+	use(15)
```
Comment (Putnam3145) [inline code/modules/power/cable.dm]:
```suggestion
	if(!use(15))
	    to_chat(user, "<span class='warning'>You need at least 15 lengths of cable!</span>
```

**Review by Putnam3145 — CHANGES_REQUESTED** [review]

File: code/modules/power/cable.dm [hunk 3]
```diff
@@ -556,23 +556,18 @@ By design, d1 is the smallest direction and d2 is the highest
...
-		to_chat(user, "<span class='notice'>You make some restraints out of cable</span>")
-		return
-	to_chat(user, "<span class='notice'>You fail to make cable restraints, you need to stand still while doing so.</span>")
-
-/obj/item/stack/cable_coil/proc/check_cable_amount(user)
-	if(amount < 15) //We dont care about cyborgs here, so we dont use get_amount()
-		to_chat(user, "<span class='notice'>You dont have enough cable coil to make restraints out of them</span>")
-		return FALSE
-	return TRUE
+    if(!use(15))
```
Comment (Putnam3145) [inline code/modules/power/cable.dm]: you're using spaces here. we use tabs

**Review by Putnam3145 — CHANGES_REQUESTED** [review]

**Review by Putnam3145 — APPROVED** [review]

**Pull request merged**